pkg/kds/global/components.go

```diff
@@ -115,16 +116,17 @@ func Setup(rt runtime.Runtime) error {
 	onGlobalToZoneSyncConnect := mux.OnGlobalToZoneSyncConnectFunc(func(stream mesh_proto.KDSSyncService_GlobalToZoneSyncServer, errChan chan error) {
 		zoneID, err := util.ClientIDFromIncomingCtx(stream.Context())
 		if err != nil {
-			errChan <- err
+			channels.NonBlockingErrorWrite(errChan, err)
```

Comment:
This is confusing I find.

Folks used to go understand the pattern:

```go
select {
case <-channel:
  // Do something
default:
  // Do something
```

I think we should just keep it as is.
A more generic question is how come we have this issue in the first place?

Reply:
we are writing to channel [here](https://github.com/kumahq/kuma/blob/ee20a34318dcd497070c2dbdf55cac932260fe0e/pkg/kds/global/components.go#L127) and none is listening on this channel because we've caught context cancelled and exited [here](https://github.com/kumahq/kuma/blob/ee20a34318dcd497070c2dbdf55cac932260fe0e/pkg/kds/mux/zone_sync.go#L88)

Reply:
moved out of util function

Reply:
maybe instead of `default:` we should check if the context is canceled?

Reply:
> maybe instead of default: we should check if the context is canceled?

but there still can be a race, we can check and see that context was not cancelled, and it can be cancelled in the meantime and still write to channel that noone is listening

Reply:
indeed. the only way to be sure is for the reader to communicate back to the writer that it's no longer listening.

Reply:
isn't it overengineering in this case? Isn't just non-blocking send of error (fire and forget) enough?

Reply:
yeah sorry, I wasn't saying we should do it. just thinking it through